src/expr.rs

```diff
@@ -722,6 +742,20 @@ fn simplify_mul(es: &[Expr]) -> Expr {
         result.push(Expr::F64(lit));
     }
     result.sort_by(|a, b| a.partial_cmp(b).unwrap());
+    let mut i = 0;
+    while result.len() > 0 && i < result.len() - 1 {
+        if let (Expr::Var(a), Expr::Pow(pow_args)) = (
+            &result[i], &result[i+1]) {
+            if let [Expr::Var(b), Expr::F64(pow_num)] = pow_args.as_slice() {
+                if a == b && *pow_num == -1.0 {
+                    result.remove(i);
+                    result.remove(i);
```

Comment:
If this branch is taken, you remove `i` twice. Shouldn't the next term to scrutinize be `i` _again_, not `i+1`?
Consider:
1. `a a^-1 b b^-1`, `i=0`, contracts to `b b^-1`
2. `b b^-1`, `i=1`, terminates as `i+1>=2`

Reply:
Good catch

Reply:
It's fixed in main.